components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m

```diff
@@ -0,0 +1,146 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCBottomDrawerPresentationController.h"
+
+#import "MDCBottomDrawerViewController.h"
+#import "private/MDCBottomDrawerContainerViewController.h"
+
+static UIColor *MDCBottomDrawerOverlayBackgroundColor(void) {
+  return [UIColor colorWithWhite:0 alpha:0.4f];
+}
+
+@interface MDCBottomDrawerPresentationController () <UIGestureRecognizerDelegate>
+
+/**
+ A semi-transparent view that darkens the visible main view when the drawer is displayed.
+ */
+@property(nonatomic) UIView *dimmingView;
+
+/**
+ The bottom drawer container view controller.
+ */
+@property(nonatomic) MDCBottomDrawerContainerViewController *bottomDrawerContainerViewController;
+
+@end
+
+@implementation MDCBottomDrawerPresentationController
+
+- (UIView *)presentedView {
+  return self.bottomDrawerContainerViewController.view;
+}
+
+- (void)presentationTransitionWillBegin {
+  UIView *containerView = [self containerView];
+
+  MDCBottomDrawerContainerViewController *bottomDrawerContainerViewController =
+      [[MDCBottomDrawerContainerViewController alloc]
+          initWithOriginalPresentingViewController:self.presentingViewController
+                                trackingScrollView:self.trackingScrollView];
+  if ([self.presentedViewController isKindOfClass:[MDCBottomDrawerViewController class]]) {
+    MDCBottomDrawerViewController *bottomDrawerViewController =
+        (MDCBottomDrawerViewController *)self.presentedViewController;
+    bottomDrawerContainerViewController.mainContentViewController =
+        bottomDrawerViewController.mainContentViewController;
+    bottomDrawerContainerViewController.headerViewController =
+        bottomDrawerViewController.headerViewController;
+  } else {
+    bottomDrawerContainerViewController.mainContentViewController = self.presentedViewController;
+  }
+  bottomDrawerContainerViewController.animatingPresentation = YES;
+  self.bottomDrawerContainerViewController = bottomDrawerContainerViewController;
+
+  self.dimmingView = [[UIView alloc] initWithFrame:self.containerView.bounds];
+  self.dimmingView.backgroundColor = MDCBottomDrawerOverlayBackgroundColor();
+  self.dimmingView.translatesAutoresizingMaskIntoConstraints = NO;
```

Comment:
Do we want this if we have autoresizingMask set to flexiblewidth and flexibleheight?

Reply:
done